src/main/java/net/dries007/tfc/client/TFCColors.java

```diff
@@ -125,25 +137,61 @@ public static int getWaterFogColor(CommonLevelAccessor level, Biome biome, Block
         return TFCBiomes.hasExtension(level, biome) ? getClimateColor(WATER_FOG_COLORS_CACHE, pos) : biome.getWaterFogColor();
     }
 
-    public static int getSeasonalFoliageColor(@Nullable BlockPos pos, int tintIndex)
+    public static int getSeasonalFoliageColor(@Nullable BlockPos pos, int tintIndex, BlockState state)
     {
-        if (pos != null && tintIndex == 0)
+        final Level level = ClientHelpers.getLevel();
+        if (level != null && pos != null)
         {
-            switch (getAdjustedNoisySeason(pos))
-            {
-                case SPRING:
-                case SUMMER:
-                    return getClimateColor(FOLIAGE_COLORS_CACHE, pos);
-                case FALL:
-                    int index = Helpers.hash(91273491823412341L, pos);
-                    return FOLIAGE_FALL_COLORS_CACHE[index & COLORMAP_MASK];
-                case WINTER:
-                    return getClimateColor(FOLIAGE_WINTER_COLORS_CACHE, pos);
-            }
+            return getSeasonalFoliageColor(pos, tintIndex, level, state);
+        }
+        return -1;
+    }
+
+    public static int getSeasonalFoliageColor(BlockPos pos, int tintIndex, LevelAccessor level, BlockState state)
+    {
+        if (tintIndex == 0)
+        {
+            return getSeasonalFoliageColor(pos, level, state);
         }
         return -1;
     }
 
+    /**
+     * Gets a color based on average temperature and time of year. Autumn occurs at different times of the year at height-adjusted average temperatures from the poles to 14c
+     */
+    private static int getSeasonalFoliageColor(BlockPos pos, LevelAccessor level, BlockState state)
```

Comment:
I'm suspicious about this being state sensitive but it seems fine.

Reply:
- Remove `BlockState state` parameter and replace with `int autumnIndex`
- Remove the calculation here:

```
final int autumnIndex = block instanceof TFCLeavesBlock ? ((TFCLeavesBlock) block).getAutumnIndex() : 127;
```